enterprise/server/remote_execution/snaploader/snaploader_test.go

```diff
@@ -81,67 +231,133 @@ func TestPackAndUnpackChunkedFiles(t *testing.T) {
-	cowA, err := copy_on_write.ConvertFileToCOW(env.GetFileCache(), originalImagePath, chunkSize, chunkDirA)
+	cowA, err := copy_on_write.ConvertFileToCOW(ctx, env, originalImagePath, chunkSize, workDirA, "")
 	require.NoError(t, err)
-	// Overwrite a random range to simulate the disk being written to. This
-	// should create some dirty chunks.
 	writeRandomRange(t, cowA)
-	// Now store a snapshot for VM A, including the COW we created.
-	taskA := &repb.ExecutionTask{}
-	keyA, err := snaploader.NewKey(taskA, "config-hash-a", "")
+	task := &repb.ExecutionTask{}
+	key, err := snaploader.NewKey(task, "config-hash", "")
 	require.NoError(t, err)
 	optsA := makeFakeSnapshot(t, workDirA)
 	optsA.ChunkedFiles = map[string]*copy_on_write.COWStore{
 		"scratchfs": cowA,
 	}
-	err = loader.CacheSnapshot(ctx, keyA, optsA)
-	require.NoError(t, err)
-	// Note: we'd normally close cowA here, but we keep it open so that
-	// mustUnpack() can verify the original contents.
-
-	// Read the bytes from cowA now to avoid relying on cowA being immutable
-	// (though it should be).
-	scratchfsBytesA := mustReadStore(t, cowA)
-
-	// Now unpack the snapshot for use by VM B, then make a modification.
-	workDirB := testfs.MakeDirAll(t, workDir, "VM-B")
-	unpackedB := mustUnpack(t, ctx, loader, keyA, workDirB, optsA)
-	cowB := unpackedB.ChunkedFiles["scratchfs"]
-	writeRandomRange(t, cowB)
-
-	// Unpack the snapshot again for use by VM C. It should see the original
-	// contents, not the modification made by VM B.
-	workDirC := testfs.MakeDirAll(t, workDir, "VM-C")
-	unpackedC := mustUnpack(t, ctx, loader, keyA, workDirC, optsA)
-	// mustUnpack already verifies the contents against cowA, but this will give
-	// us false confidence if cowA was somehow mutated. So check again against
-	// the originally snapshotted bytes, rather than the original COW instance.
-	r, err := interfaces.StoreReader(unpackedC.ChunkedFiles["scratchfs"])
-	require.NoError(t, err)
-	scratchfsBytesC, err := io.ReadAll(r)
-	require.NoError(t, err)
-	if !bytes.Equal(scratchfsBytesA, scratchfsBytesC) {
-		require.FailNow(t, "scratchfs bytes for VM C should match original contents from snapshot A")
+	err = loader.CacheSnapshot(ctx, key, optsA)
+	require.NoError(t, err)
+
+	// Delete some artifacts from the remote cache
+	snapMetadata, err := loader.GetSnapshot(ctx, key)
+	require.NoError(t, err)
+	for i, f := range snapMetadata.GetFiles() {
+		if i == 0 {
+			rn := digest.NewResourceName(f.GetDigest(), "", rspb.CacheType_CAS, repb.DigestFunction_BLAKE3).ToProto()
+			err = env.GetCache().Delete(ctx, rn)
+			require.NoError(t, err)
+			break
+		}
+	}
+	for _, f := range snapMetadata.GetChunkedFiles() {
+		for i, c := range f.GetChunks() {
+			if i == 0 {
+				rn := digest.NewResourceName(c.GetDigest(), "", rspb.CacheType_CAS, repb.DigestFunction_BLAKE3).ToProto()
+				err = env.GetCache().Delete(ctx, rn)
+				require.NoError(t, err)
+				break
+			}
```

Comment:
since you're breaking on `i == 0` (first iteration) you could remove the `i == 0` check and break unconditionally at the end of the loop. Same for the above loop